### PR TITLE
Entitlement refactoring

### PR DESCRIPTION
**Pull request by rabbah**
The refactoring in this pull request are for 
1. api keys for activations/namespaces (first commit)
~2. enabling packaged actions in rules (second commit)~
~3. bulk authorization checks for sequences (second commit)~

Also fixed a related authorization issue from issue #1470 (third commit)

### CODE REVIEW

File: core/controller/src/main/scala/whisk/core/entitlement/Entitlement.scala [hunk 1]
```diff
@@ -168,71 +187,81 @@ protected[core] abstract class EntitlementService(config: WhiskConfig, loadBalan
...
+     * Limits activations if subject exceeds limit of concurrent invocations.
+     * If the requested right is an activation, the set of resources must contain an activation of an action to be throttled.
+     * While it is possible for the set of resources to contain more than one action, the plurality is ignored and treated
+     * as one activation since these should originate from a single macro resources (e.g., a sequence).
+     */
+    protected def checkConcurrentUserThrottle(subject: Subject, right: Privilege, resources: Set[Resource])(implicit transid: TransactionId) = {
         def userThrottled = {
-            val isInvocation = resource.collection.path == Collection.ACTIONS
-            (isInvocation && !concurrentInvokeThrottler.check(subject))
+            val isInvocation = resources.filter { _.collection.path == Collection.ACTIONS }
```
Comment (markusthoemmes) [inline core/controller/src/main/scala/whisk/core/entitlement/Entitlement.scala]: `resources.exists(_.collection.path == Collection.ACTIONS)`?

**Comment by markusthoemmes** [comment]
LGTM

**Comment by markusthoemmes** [comment]
PG1 797 approved

**Comment by rabbah** [comment]
I dropped the second commit and will submit that as part of a separate entitlement related PR that completes the checks for referenced entities. This keep this pr more focused on the IAM separation.

New pg pending.

**Comment by rabbah** [comment]
new pg2/546 approved.
this should be merged after #1486.

**Comment by rabbah** [comment]
Merged as https://github.com/openwhisk/openwhisk/commit/7976c8d00377a9ca0ec2c64355339c74401014f5.

**Pull request closed without merging**